src/components/header.tsx

```diff
@@ -14,7 +14,7 @@ interface Props {
   darkModeController?: DarkModeController;
 }
 
-const Header = ({ darkModeController }: Props) => (
+const Header = ({ darkModeController }: Props): JSX.Element => (
```

Comment:
I would recommend typing this as:

```typescript
const Header: React.FC<Props> = ({ darkModeController }) => (
```

Not only is it a few characters shorter, but it also tells that the `Header` is a React functional component, instead of just being a function that happens to conform to the interface of a functional component ☺️

Reply:
Id welcome any guidance on typings with React, I dont actually know much about this at all honestly. 

What Im seeing here is that because we are using a function expression we should document that expression returns a react functional component function, instead of just documenting the return type of that function itself. 

I dont fully understand the benefits of this but I do think I understand the semantic difference.

Thank you for taking a look at this and bringing it up, Im very open to a discussion around the conventions we are using ❤️

Reply:
Id welcome any guidance on typings with React, I dont actually know much about this at all honestly. 

What Im seeing here is that because we are using a function expression we should document that expression returns a react functional component function, instead of just documenting the return type of that function itself. 

I dont fully understand the benefits of this but I do think I understand the semantic difference.

Thank you for taking a look at this and bringing it up, Im very open to a discussion around the conventions we are using ❤️

Reply:
Thanks for the feedback @LinusU

Reply:
Thanks @LinusU… react continues to impress me on how they fuzz TypeScript types :)

So I dug this out — https://unpkg.com/browse/@types/react@16.9.22/index.d.ts#L515:

```ts



type FC<P = {}> = FunctionComponent<P>;
interface FunctionComponent<P = {}> {
  (props: PropsWithChildren<P>, context?: any): ReactElement \| null;
  propTypes?: WeakValidationMap<P>;
  contextTypes?: ValidationMap<any>;
  defaultProps?: Partial<P>;
  displayName?: string;
}
```

Sure enough they wrapped an interface with a type as an alias, and sadly that means it will not type `(props: P) => JSX.Element` which is really the developer facing type in all of this — sorry, deep breaths :)

@jonchurch here is a good ref for react typing needs — https://github.com/typescript-cheatsheets/react-typescript-cheatsheet#the-types-i-need-werent-exported